File: scratch/search_links.js
const fs = require('fs');
const path = require('path');

const ROOT = 'd:\\GROWPHIL PROJECT\\MEDINGEN WRAP\\next-ssr';

const relativePatterns = [
  /href=["'](?!https?:|mailto:|tel:|\/|#)([^"']+)["']/g,
  /to=["'](?!https?:|mailto:|tel:|\/|#)([^"']+)["']/g,
  /src=["'](?!https?:|mailto:|tel:|\/|#)([^"']+)["']/g
];

function walk(dir) {
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat && stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.next') {
        walk(fullPath);
      }
    } else {
      if (file.endsWith('.js') || file.endsWith('.jsx') || file.endsWith('.ts') || file.endsWith('.tsx')) {
        checkFile(fullPath);
      }
    }
  });
}

function checkFile(filePath) {
  try {
    const content = fs.readFileSync(filePath, 'utf-8');
    const lines = content.split('\n');
    lines.forEach((line, idx) => {
      relativePatterns.forEach(pattern => {
        let match;
        // reset regex state
        pattern.lastIndex = 0;
        while ((match = pattern.exec(line)) !== null) {
          console.log(`${filePath}:${idx + 1} -> Found: ${match[0]}`);
        }
      });
    });
  } catch (err) {
    // ignore
  }
}

walk(ROOT);
console.log("Done checking.");

File: scratch/search_links.py
import os
import re

ROOT = r"d:\GROWPHIL PROJECT\MEDINGEN WRAP\next-ssr"

relative_patterns = [
    re.compile(r'href=["\'](?!https?:|mailto:|tel:|/|#)([^"\']+)["\']'),
    re.compile(r'to=["\'](?!https?:|mailto:|tel:|/|#)([^"\']+)["\']'),
    re.compile(r'src=["\'](?!https?:|mailto:|tel:|/|#)([^"\']+)["\']'),
]

for root, dirs, files in os.walk(ROOT):
    if "node_modules" in dirs:
        dirs.remove("node_modules")
    if ".next" in dirs:
        dirs.remove(".next")
        
    for file in files:
        if file.endswith((".js", ".jsx", ".ts", ".tsx")):
            path = os.path.join(root, file)
            try:
                with open(path, "r", encoding="utf-8") as f:
                    content = f.read()
                for line_no, line in enumerate(content.splitlines(), 1):
                    for pattern in relative_patterns:
                        for match in pattern.finditer(line):
                            print(f"{path}:{line_no} -> Found: {match.group(0)}")
            except Exception as e:
                pass
